extension/e2e-tests/freighterApiIntegration.test.ts

```diff
@@ -0,0 +1,315 @@
+
+  // open a second tab and go to docs playground
+  const pageTwo = await page.context().newPage();
+  await pageTwo.waitForLoadState();
+
+  const txPopupPromise = page.context().waitForEvent("page");
+
+  await pageTwo.goto(
+    "https://docs.freighter.app/docs/playground/signTransaction",
+  );
+  await pageTwo.getByRole("textbox").first().fill(TX_TO_SIGN);
+  await pageTwo
+    .getByRole("textbox")
+    .nth(1)
+    .fill("Test SDF Network ; September 2015");
+  await pageTwo.getByText("Sign Transaction XDR").click();
+
+  const txPopup = await txPopupPromise;
+
+  await expect(
+    txPopup.getByText(
+      "docs.freighter.app is currently not connected to Freighter",
+    ),
+  ).toBeVisible();
+  await expectPageToHaveScreenshot({
+    page: txPopup,
+    screenshot: "domain-not-allowed.png",
+  });
+});
+
+test("should sign auth entry when allowed", async ({ page, extensionId }) => {
+  await loginToTestAccount({ page, extensionId });
+  await allowDapp({ page });
+
+  // open a second tab and go to docs playground
+  const pageTwo = await page.context().newPage();
+  await pageTwo.waitForLoadState();
+
+  const popupPromise = page.context().waitForEvent("page");
+  await pageTwo.goto(
+    "https://docs.freighter.app/docs/playground/signAuthEntry",
+  );
+  await pageTwo.getByRole("textbox").first().fill(AUTH_ENTRY_TO_SIGN);
+  await pageTwo
+    .getByRole("textbox")
+    .nth(1)
+    .fill("Test SDF Network ; September 2015");
+  await pageTwo.getByText("Sign Authorization Entry XDR").click();
+
+  const popup = await popupPromise;
+
+  await expect(popup.getByText("Authorization Entry").first()).toBeVisible();
+  await expectPageToHaveScreenshot({
+    page: popup,
+    screenshot: "sign-auth-entry.png",
+  });
+
+  await popup.getByRole("button", { name: "Approve" }).click();
+
+  await expect(pageTwo.getByRole("textbox").nth(3)).toHaveText(
+    SIGNED_AUTH_ENTRY,
+  );
+});
+
+test("should not sign auth entry when not allowed", async ({
+  page,
+  extensionId,
+}) => {
+  await loginToTestAccount({ page, extensionId });
+
+  // open a second tab and go to docs playground
+  const pageTwo = await page.context().newPage();
+  await pageTwo.waitForLoadState();
+
+  const popupPromise = page.context().waitForEvent("page");
+  await pageTwo.goto(
+    "https://docs.freighter.app/docs/playground/signAuthEntry",
+  );
+  await pageTwo.getByRole("textbox").first().fill(AUTH_ENTRY_TO_SIGN);
+  await pageTwo
+    .getByRole("textbox")
+    .nth(1)
+    .fill("Test SDF Network ; September 2015");
+  await pageTwo.getByText("Sign Authorization Entry XDR").click();
+
+  const popup = await popupPromise;
+
+  await expect(
+    popup.getByText(
+      "docs.freighter.app is currently not connected to Freighter",
+    ),
+  ).toBeVisible();
+  await expectPageToHaveScreenshot({
+    page: popup,
+    screenshot: "domain-not-allowed.png",
+  });
+});
+
+test.skip("should sign message when allowed", async ({ page, extensionId }) => {
```

Comment:
note this test is not passing because these e2e tests caught that we're not properly passing the network in @stellar/freighter-api/signMessage

The fix is here: https://github.com/stellar/freighter/pull/1952/files#diff-a07d7b44912cd9de72ec5cbce0973a3ffa42d8f1d9d80096eb4de1efcd044300R141

I'll deploy a new version of freighter-api along with this release and that will make this test start working